src/lib.rs

```diff
@@ -101,6 +103,20 @@ pub struct GlobalCrateId {
     pub disambiguator: (u64, u64),
 }
 
+/// Where a crate came from on the local filesystem. One of these three options
+/// must be non-None.
+///
+/// Mimicks CrateSource from src/librustc/middle/cstore.rs (only paths)
+/// TODO: Uses String to be easily debuggable, change to PathBuf
+#[cfg_attr(feature = "serialize-serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serialize-rustc", derive(RustcDecodable, RustcEncodable))]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CrateSource {
+    pub dylib: Option<String>,
+    pub rlib: Option<String>,
+    pub rmeta: Option<String>,
```

Comment:
What exactly goes in these paths? Having the full path doesn't seem useful since we'd need to relocate, so I guess just the filenames. Will these always be the same when built on a different machine?

Why do we need to know what kind of crate was used?

Can there ever be more than one of these per crate (i.e., could this be an enum)?